src/m365/aad/commands/group/group-user-list.ts

```diff
@@ -0,0 +1,243 @@
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName', 'userPrincipalName', 'roles'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupDisplayName: typeof args.options.groupDisplayName !== 'undefined',
+        role: typeof args.options.role !== 'undefined',
+        properties: typeof args.options.properties !== 'undefined',
+        filter: typeof args.options.filter !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: "-i, --groupId [groupId]"
+      },
+      {
+        option: "-n, --groupDisplayName [groupDisplayName]"
+      },
+      {
+        option: "-r, --role [type]",
+        autocomplete: ["Owner", "Member"]
+      },
+      {
+        option: "-p, --properties [properties]"
+      },
+      {
+        option: "-f, --filter [filter]"
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['groupId', 'groupDisplayName']
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.role) {
+          if (['Owner', 'Member'].indexOf(args.options.role) === -1) {
+            return `${args.options.role} is not a valid role value. Allowed values Owner|Member`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const groupId = await this.getGroupId(args.options);
+
+      let users: User[] = [];
+
+      switch (args.options.role) {
+        case 'Owner':
+          users = await this.getOwners(args.options, groupId, logger);
+          break;
+        case 'Member':
+          users = await this.getMembers(args.options, groupId, logger);
+          break;
+        default:
+          const owners = await this.getOwners(args.options, groupId, logger);
+          const members = await this.getMembers(args.options, groupId, logger);
+
+          if (!args.options.properties) {
+            owners.forEach((owner: ExtendedUser) => {
+              for (let i = 0; i < members.length; i++) {
+                if (members[i].userPrincipalName === owner.userPrincipalName) {
+                  if (!owner.roles.includes('Member')) {
+                    owner.roles.push('Member');
+                  }
+                }
+              }
+            });
+          }
+
+          users = owners.concat(members);
```

Comment:
Won't this make sure members who are owners are effectively added twice?

Reply:
Hi @martinlingstuyl 

The statement below it will make sure user will added only once.
```
users = users.filter((value, index, array) => index === array.findIndex(item => item.id === value.id));
```